lib/src/mt_util.c

```diff
@@ -8,17 +8,55 @@
 #include "mt_main.h"
 
 #ifdef MTL_HAS_ASAN
-// #define MT_ASAN_MALLOC_PRINT
-/* additional memleak check for rte_malloc since dpdk asan not support this */
-static int g_mt_rte_malloc_cnt;
-
+#include <execinfo.h>
+
+#define MAX_BT_SIZE 32
+/* backtrace info */
+struct mt_backtrace_info {
+  void* pointer;
+  size_t size;
+  char** bt_strings;
+  int bt_size;
+  MT_TAILQ_ENTRY(mt_backtrace_info) next;
+};
+
+/* List of backtrace info */
+MT_TAILQ_HEAD(mt_backtrace_info_list, mt_backtrace_info);
+struct mt_backtrace_info_list g_bt_head;
+pthread_mutex_t g_bt_mutex;
+
+/* additional memleak check for mempool_create since dpdk asan not support this */
 static int g_mt_mempool_create_cnt;
 
+int mt_asan_init(void) {
+  mt_pthread_mutex_init(&g_bt_mutex, NULL);
+  MT_TAILQ_INIT(&g_bt_head);
+  return 0;
+}
+
 int mt_asan_check(void) {
-  if (g_mt_rte_malloc_cnt != 0) {
-    rte_panic("%s, detect not free memory by rte_malloc, leak cnt %d\n", __func__,
-              g_mt_rte_malloc_cnt);
+  /* check if any not freed */
+  int leak_cnt = 0;
+  struct mt_backtrace_info* bt_info = NULL;
+  while ((bt_info = MT_TAILQ_FIRST(&g_bt_head))) {
+    info("%s, \033[31mleak of %" PRIu64 " byte(s) at %p\033[0m\n", __func__,
+         bt_info->size, bt_info->pointer);
+    if (bt_info->bt_strings) {
+      info("%s, backtrace info:\n", __func__);
+      for (int i = 0; i < bt_info->bt_size; ++i) {
+        info("%s, %s\n", __func__, bt_info->bt_strings[i]);
+      }
+      mt_free(bt_info->bt_strings);
+      bt_info->bt_strings = NULL;
+    }
+    MT_TAILQ_REMOVE(&g_bt_head, bt_info, next);
+    rte_free(bt_info->pointer);
```

Comment:
Can you add a note for why rte_free is directly used, like below?
```
rte_free(bt_info->pointer); /* use raw free directly */
```

Reply:
sure